api/lightwallet_rest.md

```diff
@@ -0,0 +1,402 @@
+| tx_id            | uint64            | Index of tx in blockchain     |
+| amount           | uint64-string     | XMR value of output           |
+| index            | uint16            | Index within vout vector      |
+| global_index     | uint64-string     | Index within amount           |
+| rct              | binary            | Bytes of ringct data          |
+| tx_hash          | binary            | Bytes of tx hash              |
+| tx_prefix_hash   | binary            | Bytes of tx prefix hash       |
+| public_key       | binary            | Bytes of output public key    |
+| tx_pub_key       | binary            | Bytes of the tx public key    |
+| spend_key_images | array of binary's | Bytes of key images           |
+| timestamp        | timestamp         | Timestamp of containing block |
+| height           | uint64            | Containing block height       |
+
+> `tx_id` is determined by the monero daemon. It is the offset that a
+> transaction appears in the blockchain from the genesis block.
+
+> `global_index` is determined by the monero daemon. It is the offset from the
+> first time the amount appeared in the blockchain. After ringct, this is the
+> order of outputs as they appear in the blockchain.
+
+> `tx_hash` and `tx_prefix_hash` are determined by how `monerod` computes the
+> hash.
+
+> `rct` is, for ringct outputs, a 96-byte blob containing the concatenation
+> of the public commitment, then the ringct mask value, and finally the
+> ringct amount value. For ringct coinbase outputs, the mask is always the
+> identity mask and the amount is zero; for non-coinbase ringct outputs, the
+> mask and amount are the respective raw encrypted values, which must be
+> decrypted by the client using the view secret key. For non-ringct outputs,
+> this field is nil.
+
+**rates** object
+
+| Field |  Type   |      Description      |
+|-------|---------|-----------------------|
+| AUD * | float32 | AUD/XMR exchange rate |
+| BRL * | float32 | BRL/XMR exchange rate |
+| BTC * | float32 | BTC/XMR exchange rate |
+| CAD * | float32 | CAD/XMR exchange rate |
+| CHF * | float32 | CHF/XMR exchange rate |
+| CNY * | float32 | CNY/XMR exchange rate |
+| EUR * | float32 | EUR/XMR exchange rate |
+| GBP * | float32 | GBP/XMR exchange rate |
+| HKD * | float32 | HKD/XMR exchange rate |
+| INR * | float32 | INR/XMR exchange rate |
+| JPY * | float32 | JPY/XMR exchange rate |
+| KRW * | float32 | KRW/XMR exhcnage rate |
+| MXN * | float32 | MXN/XMR exchange rate |
+| NOK * | float32 | NOK/XMR exchange rate |
+| NZD * | float32 | NZD/XMR exchange rate |
+| SEK * | float32 | SEK/XMR exchange rate |
+| SGD * | float32 | SGD/XMR exchange rate |
+| TRY * | float32 | TRY/XMR exchange rate |
+| USD * | float32 | USD/XMR exchange rate |
+| RUB * | float32 | RUB/XMR exchange rate |
+| ZAR * | float32 | ZAR/XMR exchange rate |
+
+> If an exchange rate is unavailable, the server field shall omit the field
```

Comment:
I think it could be useful and helpful to mention from where Monero software will take the exchange rate.

Reply:
Does it need to be from a consistent place for all implementations? This seemed restrictive.

Reply:
I was trying to ask: is there any central source for retrieving prices? I saw you put cryptocompare API in your pull request on monero-project/monero.

Reply:
I think the source of exchange rates has to remain up to the implementation. There doesn't exist a universally canonical source, and I don't see any reason to force implementations to use the same ones (or even provide them at all -- note, these fields are marked as optional.)

Reply:
I agree with you.